enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,380 @@
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Some OpenShift components (such as Etcd and Ingress) require
+a minimum of 2 or 3 nodes. The `single-node-production-edge`
+cluster profile will configure these components as appropriate
+for a single node.
+
+Known gaps:
+- `cluster-etcd-operator` will not deploy the etcd cluster without minimum of 3 master nodes (can be changed by enabling `useUnsupportedUnsafeNonHANonProductionUnstableEtcd`)
+- Even with the unsupported feature flag, `etcd-quorum-guard` still requires 3 nodes due to its replica count.
+- `cluster-authentication-operator` will not deploy `OAuthServer` without minimum of 3 master nodes (can be change by enabling `useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer`)
+- `cluster-ingress-operator` deploys the router with 2 replicas. On a single node one will fail to start and the ingress will show as degraded.
+
+When we are deploying a cluster with the `single-node-production-edge`
+cluster profile the relevant operators should support a non-HA
+configuration that makes the correct adjustments to the deployment
+(e.g., `cluster-ingress-operator` should deploy a single router,
+`cluster-etcd-operator` should deploy the `etcd-member`
+[without waiting for 3 master nodes](https://github.com/openshift/cluster-etcd-operator/blob/98590e6ecfe282735c4eff01432ae40b29f81202/pkg/etcdenvvar/etcd_env.go#L72))
+
+In addition, some components are not relevant for this cluster
+profile (e.g. console, cluster-autoscaler, marketplace?) and shouldn't
+be deployed at all.
+
+
+### Risks and Mitigations
+
+*What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.*
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads frequently require a realtime kernel. How will a
+   user specify whether to use the realtime or regular kernel? Should
+   we assume they always want the realtime version?
+2. Similarly, telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+3. The machine-config-operator works by (almost always) rebooting a host.
+   Is that going to be OK in these single-node deployments? Do we want
+   the machine-config-operator disabled? Do we want it to run in a
+   different mode where reboots are not performed?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have
+CI coverage informing the release. An end-to-end job using the profile
+and running an appropriate subset of the standard OpenShift tests
+will be created and configured to block accepting release images
+unless it passes.
```

Comment:
I would also expect an integration CI job that runs with the optional operators we expect will target this deployment topology. Launching against this environment will need to be a supported target topology in our common test infra for operators.

Additionally I would expect the ability to launch one of these single node deployments from cluster-bot